### DISCUSSION TITLE
Required dependencies on build.gradle.kts to run connector.jar on EDC repo

### DISCUSSION THREAD

**Discussion by nicktep** [eclipse-edc/Connector#4064] [Q&A]
Hello, 

I want to run a connector.jar only with required dependencies to run successfully. What should i write on build.gradle.kts to make it happen. i tried  the code from samples kts file to copy it to edc kts file but i take an error as you can see in the picture.

`plugins {
    `java-library`
    id("application")
    alias(libs.plugins.shadow)
}

dependencies {
    implementation(libs.edc.boot)
    implementation(libs.edc.connector.core)
}

application {
    mainClass.set("org.eclipse.edc.boot.system.runtime.BaseRuntime")
}

tasks.withType<com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar> {
    mergeServiceFiles()
    archiveFileName.set("basic-connector.jar")
}`

![image](https://github.com/eclipse-edc/Connector/assets/78322657/aedabbc8-8f99-49be-a5f6-3650dba08ca0)

Also i tried this code. 

`/*
 * This file was generated by the Gradle 'init' task.
 *
 * This generated file contains a sample Java application project to get you started.
 * For more details on building Java & JVM projects, please refer to https://docs.gradle.org/8.7/userguide/building_java_projects.html in the Gradle documentation.
 */

plugins {
    `java-library`
    id("application")
    id("com.github.johnrengelman.shadow") version "8.1.1"
}

repositories {
    mavenCentral()
}

// pick-up the latest version from https://github.com/eclipse-edc/Connector/releases
val edcVersion = "0.5.1"

dependencies {
    // the core control-plane module set
    implementation("org.eclipse.edc:control-plane-core:${edcVersion}")
    implementation("org.eclipse.edc:control-plane-api-client:${edcVersion}")

    // the dsp protocol module
    implementation("org.eclipse.edc:dsp:${edcVersion}")

    // a test implementation of the `IdentityService`
    implementation("org.eclipse.edc:iam-mock:${edcVersion}")

    // implementation("org.eclipse.edc.connector.api.client.spi.transferprocess.TransferProcessApiClient")

    // management api, will permit to manage the connector through a REST interface
    implementation("org.eclipse.edc:management-api:${edcVersion}")

    // the core data-plane-selector module set
    implementation("org.eclipse.edc:data-plane-selector-core:${edcVersion}")
    
    // the core data-plane module set
    implementation("org.eclipse.edc:data-plane-core:${edcVersion}")

    implementation ("org.junit.jupiter:junit-jupiter-api:5.5.1")
}

// configure the main class
application {
    mainClass.set("org.eclipse.edc.boot.system.runtime.BaseRuntime")
}

// build the application as a single jar that will be available in `build/libs/connector.jar`
tasks.withType<com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar> {
    mergeServiceFiles()
    archiveFileName.set("connector.jar")
}`

and the error i take. 

![image](https://github.com/eclipse-edc/Connector/assets/78322657/0f88b3b7-c328-458f-a6fd-071b4fcd6c76)

I don't know what the requirements to make it work. also i dont understand the errors. Can you please help me to run a basic connector inside the edc connector repo.

**Comment by paullatzelsperger** [comment]
I recommend getting familiar with 
- Gradle as a build system
- Version Catalogs
- the modularity concept of Java
- the modularity concept of EDC

I'm sure you'll understand that we are unable to provide individual support, when tutorials like the [Samples repo](https://github.com/eclipse-edc/Samples) exist.
